packages/origin.js/src/resources/purchases.js

```diff
@@ -1,48 +1,57 @@
-class Purchases {
-    constructor({ contractService, ipfsService }) {
-      this.contractService = contractService
-      this.ipfsService = ipfsService
-    }
+import ResourceBase from"../ResourceBase"
 
-    async contractFn(address, functionName, args=[], value=0){
-        const purchaseContract = this.contractService.purchaseContract
-        const purchase = await purchaseContract.at(address)
-        const account = await this.contractService.currentAccount()
-        args.push({ from: account, value: value })
-        return await purchase[functionName].apply(purchase, args)
-    }
+const _STAGES_TO_NUMBER = {
+  awaiting_payment: 0,
+  shipping_pending: 1,
+  buyer_pending: 2,
+  seller_pending: 3,
+  in_dispute: 4,
+  review_period: 5,
+  complete: 6
+}
+const _NUMBERS_TO_STAGE = {}
 
-    async get(address){
-        const contractData = await this.contractFn(
-            address,
-            'data'
-        )
-        return {
-            address: address,
-            stage: contractData[0], // perhaps return a string
-            listingAddress: contractData[1],
-            buyerAddress: contractData[2],
-            created: contractData[3],
-            buyerTimout: contractData[4]
-        }
-    }
 
-    async pay(address, amountWei){
-        return await this.contractFn(
-            address,
-            'pay',
-            [],
-            amountWei
-        )
-    }
+class Purchases extends ResourceBase{
+  constructor({ contractService, ipfsService }) {
+    super({ contractService, ipfsService })
+    
+    this.contractDefinition = this.contractService.purchaseContract
 
-    async buyerConfirmReceipt(address){
-        return await this.contractFn( address, 'buyerConfirmReceipt')
-    }
+    this.STAGES = {}
+    Object.entries(_STAGES_TO_NUMBER).map(([k, v]) => {
```

Comment:
We should be using `forEach` here, not `map`. `forEach` just iterates, while `map` returns a new array. We're not doing anything with the return value of this method call so we shouldn't be using `map`.